lib/src/recovery_protocol/recovery_protocol.dart

```diff
@@ -0,0 +1,207 @@
+      case NgScannerState.hasError:
+        return hasError(current, reader);
+      case NgScannerState.isEndOfFile:
+        return isEndOfFile(current, reader);
+      case NgScannerState.scanAfterComment:
+        return scanAfterComment(current, reader);
+      case NgScannerState.scanAfterElementDecorator:
+        return scanAfterElementDecorator(current, reader);
+      case NgScannerState.scanAfterElementDecoratorValue:
+        return scanAfterElementDecoratorValue(current, reader);
+      case NgScannerState.scanAfterElementIdentifierClose:
+        return scanAfterElementIdentifierClose(current, reader);
+      case NgScannerState.scanAfterElementIdentifierOpen:
+        return scanAfterElementIdentifierOpen(current, reader);
+      case NgScannerState.scanAfterInterpolation:
+        return scanAfterInterpolation(current, reader);
+      case NgScannerState.scanBeforeElementDecorator:
+        return scanBeforeElementDecorator(current, reader);
+      case NgScannerState.scanBeforeInterpolation:
+        return scanBeforeInterpolation(current, reader);
+      case NgScannerState.scanComment:
+        return scanComment(current, reader);
+      case NgScannerState.scanInterpolation:
+        return scanInterpolation(current, reader);
+      case NgScannerState.scanElementDecorator:
+        return scanElementDecorator(current, reader);
+      case NgScannerState.scanElementDecoratorValue:
+        return scanElementDecoratorValue(current, reader);
+      case NgScannerState.scanElementEndClose:
+        return scanElementEndClose(current, reader);
+      case NgScannerState.scanElementEndOpen:
+        return scanElementEndOpen(current, reader);
+      case NgScannerState.scanElementIdentifierClose:
+        return scanElementIdentifierClose(current, reader);
+      case NgScannerState.scanElementIdentifierOpen:
+        return scanElementIdentifierOpen(current, reader);
+      case NgScannerState.scanElementStart:
+        return scanElementStart(current, reader);
+      case NgScannerState.scanSimpleElementDecorator:
+        return scanSimpleElementDecorator(current, reader);
+      case NgScannerState.scanSpecialBananaDecorator:
+        return scanSpecialBananaDecorator(current, reader);
+      case NgScannerState.scanSpecialEventDecorator:
+        return scanSpecialEventDecorator(current, reader);
+      case NgScannerState.scanSpecialPropertyDecorator:
+        return scanSpecialPropertyDecorator(current, reader);
+      case NgScannerState.scanStart:
+        return scanStart(current, reader);
+      case NgScannerState.scanSuffixBanana:
+        return scanSuffixBanana(current, reader);
+      case NgScannerState.scanSuffixEvent:
+        return scanSuffixEvent(current, reader);
+      case NgScannerState.scanSuffixProperty:
+        return scanSuffixProperty(current, reader);
+      case NgScannerState.scanText:
+        return scanText(current, reader);
+    }
+    return new RecoverySolution.skip();
```

Comment:
Technically this part should never be hit in the first place; I just put this here to get rid of the pesky 'doesn't return with a ...' message.